test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java

```diff
@@ -0,0 +1,215 @@
+    public static void waitForPodsReady(String namespaceName, LabelSelector selector, int expectPodsCount,
+                                        boolean containers, Runnable onTimeout) {
+        Wait.until("readiness of all Pods matching: " + selector,
+                TestFrameConstants.GLOBAL_POLL_INTERVAL_MEDIUM, READINESS_TIMEOUT,
+                () -> {
+                    List<Pod> pods = KubeResourceManager.getKubeClient().getClient().pods()
+                            .inNamespace(namespaceName).withLabelSelector(selector).list().getItems();
+                    if (pods.isEmpty() && expectPodsCount == 0) {
+                        LOGGER.debug("All expected Pods {} in namespace {} are ready", selector, namespaceName);
+                        return true;
+                    }
+                    if (pods.isEmpty()) {
+                        LOGGER.debug("Pods matching {}/{} are not ready", namespaceName, selector);
+                        return false;
+                    }
+                    if (pods.size() != expectPodsCount) {
+                        LOGGER.debug("Expected Pods {}/{} are not ready", namespaceName, selector);
+                        return false;
+                    }
+                    for (Pod pod : pods) {
+                        if (!(Readiness.isPodReady(pod) || Readiness.isPodSucceeded(pod))) {
+                            LOGGER.debug("Pod is not ready: {}/{}", namespaceName, pod.getMetadata().getName());
+                            return false;
+                        } else {
+                            if (containers) {
+                                for (ContainerStatus cs : pod.getStatus().getContainerStatuses()) {
+                                    if (!(Boolean.TRUE.equals(cs.getReady())
+                                            || cs.getState().getTerminated().getReason().equals("Completed"))) {
+                                        LOGGER.debug("Container {} of Pod {}/{} not ready",
+                                                cs.getName(), namespaceName, pod.getMetadata().getName());
+                                        return false;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    LOGGER.info("Pods matching {}/{} are ready", namespaceName, selector);
+                    return true;
+                }, onTimeout);
+    }
+
+    /**
+     * Wait for pod ready, if not ready for timeout try to restart and check again
+     *
+     * @param namespaceName namespace
+     * @param selector label selector
+     * @param expectedPodsCount expected pods count
+     * @param containersReady flag wait for containers
+     */
+    public static void waitForPodsReadyWithRestart(String namespaceName, LabelSelector selector,
+                                                   int expectedPodsCount, boolean containersReady) {
+        try {
```

Comment:
Why is try catch here?

Reply:
Wait exception is thrown so it is correct... as wait exception is based on runtime ex which is not defined in method header....